### ISSUE TITLE
Command helm-projectile reports "apply: Symbol’s value as variable is void: symbolp"

### ISSUE THREAD

**Issue by huangfeiyu** [bbatsov/helm-projectile#164]
## Expected behavior

open the helm window.

## Actual behavior

show error message "apply: Symbol’s value as variable is void: symbolp"

## Steps to reproduce the problem
run command `helm-projectile`

## Backtraces if necessary (`M-x toggle-debug-on-error`)

```
Debugger entered--Lisp error: (void-variable symbolp)
  #f(compiled-function (source) #<bytecode 0x156acf8a209d>)(#<helm-source-sync helm-source-sync-156ad072185c>)
  apply(#f(compiled-function (source) #<bytecode 0x156acf8a209d>) #<helm-source-sync helm-source-sync-156ad072185c>)
  #f(compiled-function (&rest args) #<bytecode 0x156acf9443f1>)(#<helm-source-sync helm-source-sync-156ad072185c>)
  apply(#f(compiled-function (&rest args) #<bytecode 0x156acf9443f1>) #<helm-source-sync helm-source-sync-156ad072185c> nil)
  helm--setup-source(#<helm-source-sync helm-source-sync-156ad072185c>)
  helm-make-source("Projectile files" helm-source-sync :before-init-hook #f(compiled-function () #<bytecode 0x156ad0721809>) :candidates #f(compiled-function () #<bytecode 0x156ad0721825>) :filtered-candidate-transformer #f(compiled-function (files source) #<bytecode 0x156ad072184d>) :fuzzy-match t :keymap (keymap (left . helm-previous-source) (right . helm-next-source) (127 . helm-helm-ff-delete-char-backward-with-subkeys) (S-f6 . helm-ff-sort-by-ext) (S-f5 . helm-ff-toggle-files-only) (S-f4 . helm-ff-toggle-dirs-only) (S-f3 . helm-ff-sort-by-size) (S-f2 . helm-ff-sort-by-newest) (S-f1 . helm-ff-sort-alpha) (18) (31 . helm-ff-undo) (67108922 . helm-ff-complete-tramp-methods) (12) (67108911 . helm-ff-run-fd) (C-backspace . helm-ff-run-toggle-auto-update) (67108987 . helm-enlarge-window) (67108989 . helm-narrow-window) (3 keymap (97 . #f(compiled-function () ... #<bytecode 0x156acf93e139>)) (102 . #f(compiled-function () ... #<bytecode 0x156acf93e121>)) (20 . helm-ff-find-file-other-tab) (64 . helm-ff-run-insert-org-link) (114 . helm-ff-run-find-file-as-root) (47 . helm-ff-run-find-sh-command) (112 . helm-ff-run-print-file) (1 . helm-ff-run-mail-attach-files) (127 . helm-ff-run-toggle-auto-update) (104 . helm-ff-file-name-history) (61 . helm-ff-run-ediff-file) (116 . helm-ff-toggle-thumbnails) (88 . helm-ff-run-open-file-with-default-tool) (22 . helm-ff-run-preview-file-externally) (24 . helm-ff-run-open-file-externally) (15 . helm-ff-run-switch-other-frame) (111 . helm-ff-run-switch-other-window) (105 . helm-ff-run-complete-fn-at-point) (100 . helm-ff-persistent-delete)) (27 keymap (45 . helm-ff-decrease-image-size-persistent) (43 . helm-ff-increase-image-size-persistent) (114 . helm-ff-rotate-right-persistent) (108 . helm-ff-rotate-left-persistent) (105 . helm-ff-properties-persistent) (112 . helm-find-files-history) (61 . helm-ff-run-ediff-merge-file) (67108901 . helm-ff-run-query-replace-regexp) (37 . helm-ff-run-query-replace) (64 . helm-ff-run-query-replace-fnames-on-marked) (33 . #f(compiled-function () ... #<bytecode 0x156acf93e18d>)) (101 . #f(compiled-function () ... #<bytecode 0x156acf93e155>)) (84 . helm-ff-run-touch-files) (75 . helm-ff-run-kill-buffer-persistent) (68 . helm-ff-run-delete-file) (72 . helm-ff-run-hardlink-file) (89 . helm-ff-run-relsymlink-file) (83 . helm-ff-run-symlink-file) (76 . helm-ff-run-load-file) (66 . helm-ff-run-byte-compile-file) (86 . helm-ff-run-rsync-file) (67 . helm-ff-run-copy-file) (82 . helm-ff-run-rename-file) (46 . #f(compiled-function () ... #<bytecode 0x156acf93e171>)) ...) (19 . helm-ff-run-grep) (24 keymap (22 . helm-ff-run-find-alternate-file) (17 . helm-ff-run-marked-files-in-dired) (114 keymap ... ...) (4 . helm-ff-run-browse-project) (6 . helm-ff-run-locate)) (29 . helm-ff-run-toggle-basename) (13 . helm-ff-RET) keymap (f12 . #f(compiled-function () ... #<bytecode 0x156acf8a39c9>)) ...) :help-message helm-ff-help-message :mode-line "\\<helm-read-file-map>\\[hel..." :action (("Find File" . helm-find-file-or-marked) ("Find file in Dired" . helm-point-file-in-dired) ("View file" . view-file) ("Query replace fnames on ma..." . helm-ff-query-replace-fnames-on-marked) ("Marked files in dired `C-x..." . helm-marked-files-in-dired) ("Query replace contents on ..." . helm-ff-query-replace) ("Query replace regexp conte..." . helm-ff-query-replace-regexp) ("Attach file(s) to mail buf..." . helm-ff-mail-attach-files) ("Serial rename files" . helm-ff-serial-rename) ("Serial rename by symlinkin..." . helm-ff-serial-rename-by-symlink) ("Serial rename by copying f..." . helm-ff-serial-rename-by-copying) ("Open file with default too..." . helm-open-file-with-default-tool) ("Find file in hex dump" . hexl-find-file) ("Insert as org link `C-c @'" . helm-files-insert-as-org-link) ("Fd shell command (C-/)" . helm-ff-fd) ("Find files in file" . helm-find-files-in-file) ("Open file externally `C-c ..." . helm-open-file-externally) ("Grep File(s) `C-s, C-u Rec..." . helm-find-files-grep) ("Grep current directory wit..." . helm-find-files-ag) ("Git grep `M-g g, C-u from ..." . helm-ff-git-grep) ("Zgrep File(s) `M-g z, C-u ..." . helm-ff-zgrep) ("Pdf Grep File(s)" . helm-ff-pdfgrep) ("Gid `M-g i'" . helm-ff-gid) ("Switch to Eshell `M-e'" . helm-projectile-switch-to-shell) ("Etags `M-., C-u reload tag..." . helm-projectile-ff-etags-select-action) ("Eshell command on file(s) ..." . helm-projectile-find-files-eshell-command-on-file-action) ...) :persistent-action helm-projectile-file-persistent-action :persistent-help "Preview file")
  (defvar helm-source-projectile-files-list (helm-make-source "Projectile files" 'helm-source-sync :before-init-hook #f(compiled-function () #<bytecode 0x156ad0721809>) :candidates #f(compiled-function () #<bytecode 0x156ad0721825>) :filtered-candidate-transformer #f(compiled-function (files source) #<bytecode 0x156ad072184d>) :fuzzy-match helm-projectile-fuzzy-match :keymap helm-projectile-find-file-map :help-message 'helm-ff-help-message :mode-line helm-read-file-name-mode-line-string :action helm-projectile-file-actions :persistent-action 'helm-projectile-file-persistent-action :persistent-help "Preview file") ("/home/huang/.emacs.d/elpa/helm-projectile-20220807..." . 20783))
  autoload-do-load((autoload "helm-projectile" "Use projectile with Helm instead of ido.\n\nWith a p..." t nil) helm-projectile)
  command-execute(helm-projectile)

```

## Environment & version information

- `helm-projectile` version: helm-projectile-20220807.350
- `helm` version (in `helm-pkg.el`): helm-20220803.1447
- `projectile` version (`M-x projectile-version`): Projectile 20220804.1530
- Emacs version (`M-x emacs-version`): GNU Emacs 27.1 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.24, cairo version 1.16.0) of 2021-03-28, modified by Debian
- OS: debian 11

Thanks!

**Comment by WillForan** [comment]
in case it's helpful. here's the change in helm 6 days ago
https://github.com/emacs-helm/helm/commit/00bf87a6382dab67891c585869df546e1d753827#diff-576bddbe5b6638c1f22677eb338694bc851c549c148577ce3e3efebb1e36587cR3725

**Comment by kczulko** [comment]
@thierryvolpiatto @WillForan ,
I am an emacs/elisp noob, can you please suggest some kind of workaround for this error?

**Comment by thierryvolpiatto** [comment]
kczulko ***@***.***> writes:

> @thierryvolpiatto @WillForan ,
> I am an emacs/elisp noob, can you please suggest some kind of workaround for this error?

Looks like pcase in emacs-27.1 doesn't support yet (pred (not <predicate>)).
Should be fixed by 1d066466 in helm master.

Thanks.

> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you were mentioned.*Message ID: ***@***.***>


-- 
Thierry

**Comment by kczulko** [comment]
For now as a dummy workaround I've replaced all the helm-core elisp files with those from https://github.com/emacs-helm/helm/tree/v3.8.6 which works in my case.

Thank you @thierryvolpiatto for all the work you do.

**Comment by thierryvolpiatto** [comment]
kczulko ***@***.***> writes:

> For now as a dummy workaround I've replaced all the helm-core elisp
> files with those from https://github.com/emacs-helm/helm/tree/v3.8.6
> which works in my case.

Please update to last helm which should fix your emacs-27.1 bug.

Thanks.

-- 
Thierry

**Comment by huangfeiyu** [comment]
Thank you all!
